cluster-autoscaler/estimator/pod_priority_processor.go

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package estimator
+
+import (
+	"sort"
+
+	apiv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/kubernetes/pkg/scheduler/framework"
+)
+
+// PodScoreInfo contains Pod and score that corresponds to how important it is to handle the pod first.
+type PodScoreInfo struct {
+	score float64
+	pod   *apiv1.Pod
+}
+
+// PodPriorityProcessor is an interface used to determine the order of the pods
+// used while binpacking during scale up estimation
+type PodPriorityProcessor interface {
+	Process(pods []*apiv1.Pod, nodeTemplate *framework.NodeInfo) []*apiv1.Pod
+}
+
+// DefaultPodPriorityProcessor is the default implementation of the pod priority processor
+// It refactors the existing implementation of sorting pods by pod score
+type DefaultPodPriorityProcessor struct {
```

Comment:
"Default" doesn't tell anything about the behavior of the type, could we name it according to the behavior? E.g. something like "DecreasingPodOrderer" (to match the intent of it implementing First-fit-decreasing binpacking: https://en.wikipedia.org/wiki/First-fit-decreasing_bin_packing).

Reply:
Yeah. I think better naming would be more appropriate. Done.